src/hooks/miniBackoffice.ts

```diff
@@ -143,6 +145,9 @@ export const useMiniBackoffice = (
       registerUser: reportTransactionTo(setFractalId, () =>
         selfServeRegistryOperator.connect(signer).addSelf(keccak256(account))
       ),
+      approveUser: reportTransactionTo(setKycStatus, () =>
+        selfServeRegistryOperator.connect(signer).addSelfToList(KYCList)
+      ),
```

Comment:
Please change: You can't approve the user until it exists. Adding this function on `UnregisteredUser` breaks the API design of [making impossible transitions unrepresentable](https://fsharpforfunandprofit.com/posts/designing-with-types-making-illegal-states-unrepresentable/).

The obvious alternative here, for me, would be to return a `registerAndApproveUser`. I know it feels clunky, but it maintains the design intent and fulfils your use case.

Another alternative would be to complicate things a little bit on `src/components/ActionPage/MiniBackoffice.tsx`. This API would be kept tidier. However, even though I didn't take a stab at it, I don't think the complexity trade-off is worth it.

Reply:
Just for clarification. The goal of this PR (and associated ticket) is to be able to show to the sales people how it would work by combining 2 steps into 1. If they liked it better than the existing 2-step, then we would need to implement a method in smart contract (as well as more changes to the app) because this PR still runs two separate transactions through the wallet. 

Just so that I understand, you're saying I should ..
1. In the miniBackOffce.tsx file create two new interfaces, one for RegisterAndApproveUser but that's its ok to leave the two methods disapproveUser() and unRegisterUser in the KYCApproved interface?
2. Am I Replace the return value in the hook itself for RegisterUser with RegisterAndApproveUser but leave the return value for KYCApproved?

Reply:
I was having trouble using english to explain myself, so I just committed something: https://github.com/trustfractal/did-registry-demo-dapp/pull/7/commits/c06c62ee83a8a83dc195452396537ca24643903f